### ISSUE TITLE
CA1001 fires for HttpClient

### ISSUE THREAD

**Issue by bc3tech** [dotnet/roslyn-analyzers#7584]
### Analyzer

[CA1001](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1001): `Types that own disposable fields should be disposable`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100](https://dotnet.microsoft.com/download/dotnet/8.0)

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0

### Steps To Reproduce

```csharp
internal sealed class Foo() // Analyzer fires on this class and it should not
{
    private readonly HttpClient _httpClient = new();
}
```

This guidance is per https://github.com/mspnp/performance-optimization/tree/main/ImproperInstantiation

### Expected behavior

Analyzer should not fire if the only IDisposable obj is HttpClient
